pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/ClientCredentialsFlow.java

```diff
@@ -46,27 +46,39 @@
 class ClientCredentialsFlow extends FlowBase {
     public static final String CONFIG_PARAM_ISSUER_URL = "issuerUrl";
     public static final String CONFIG_PARAM_AUDIENCE = "audience";
+    // Maps to the keyFileUrl
     public static final String CONFIG_PARAM_KEY_FILE = "privateKey";
     public static final String CONFIG_PARAM_SCOPE = "scope";
 
     private static final long serialVersionUID = 1L;
 
     private final String audience;
-    private final String privateKey;
+    private final String keyFileUrl;
```

Comment:
`keyFileUrl` seems repetitive and conflicting. Is it a File or an URL? What key is it? Perhaps keeping the name `privateKey` and switching to use types would improve clarity. I'd recommend `java.net.URI` class also for URLs. that would make this `URI privateKey`.